### PR TITLE
Improve Route and Bus Schedule Features

### PR DESCRIPTION
**Pull request by ganilala**

### CODE REVIEW

File: pentasharp/Controllers/BusSchedule.cs [hunk 1]
```diff
@@ -30,24 +31,44 @@ public BusScheduleController(AppDbContext context, IMapper mapper)
...
+                                string.IsNullOrWhiteSpace(model.FromLocation) ||
+                                string.IsNullOrWhiteSpace(model.ToLocation);
 
-            if (_context.BusRoutes.Any(r => r.FromLocation == model.FromLocation && r.ToLocation == model.ToLocation))
-                return BadRequest(new { success = false, message = "Route already exists." });
+            if (isInvalidData)
+            {
+                return BadRequest(new
+                {
+                    success = false,
```
Comment (GentaSL) [inline pentasharp/Controllers/BusSchedule.cs]: By standardized response i meant:
Create a separate class thats named something like StandardResponse what would look similar: 
public class StandardResponse<T>
{
    public bool Success { get; set; }
    public string Message { get; set; }
    public string Code { get; set; }
    public T Data { get; set; }
}

crate an Enum to define common codes and map them in the messages 
public enum ResponseCodes
{
    Success,
    InternalServerError,
    InvalidData,
    NotFound
}

public static class ResponseMessages
{
    public const string InternalServerError = "An unexpected error occurred.";
    public const string InvalidData = "Invalid data provided.";
    public const string NotFound = "Resource not found.";
}

Implement a generic factory method that uses the enums to generate responses dynamically, and then use it similarly to
if (isInvalidData)
{
    var errorResponse = BuildProviderResponse<object>(ResponseCodes.InvalidData);
    return BadRequest(errorResponse); // Wrap in BadRequest to send as HTTP 400
}
 
 fix this approach throughout the places I left that comment. 
 Note: dont just copy and paste what i said, check if you agree on it being a better approach

File: pentasharp/Controllers/BusSchedule.cs [hunk 2]
```diff
@@ -63,28 +84,56 @@ public IActionResult GetRoutes()
         [HttpPut("EditRoute/{id}")]
         public IActionResult EditRoute(int id, [FromBody] AddRouteViewModel model, int hours, int minutes)
         {
-            var route = _context.BusRoutes.Find(id);
+            var route = _context.BusRoutes.SingleOrDefault(r => r.RouteId == id);
```
Comment (GentaSL) [inline pentasharp/Controllers/BusSchedule.cs]: if routeId is a primary key, i guess its better for us to stick with Find, if not keep this approach

File: pentasharp/Controllers/BusSchedule.cs [hunk 3]
```diff
@@ -264,9 +397,9 @@ public IActionResult GetReservations()
                         r.Schedule.ArrivalTime,
                         r.Schedule.Price,
                         r.Schedule.AvailableSeats,
-                        r.Schedule.Bus.BusNumber,
-                        r.Schedule.Route.FromLocation,
-                        r.Schedule.Route.ToLocation
+                        BusNumber = r.Schedule.Bus.BusNumber,
+                        FromLocation = r.Schedule.Route.FromLocation,
+                        ToLocation = r.Schedule.Route.ToLocation
```
Comment (GentaSL) [inline pentasharp/Controllers/BusSchedule.cs:531]: consistency, either similar approach to this  BusNumber = r.Schedule.Bus.BusNumber, in all parameters or the old approach r.Schedule.Bus.BusNumber,

File: pentasharp/Controllers/BusSchedule.cs (hunk 1 above)
Comment (ganilala) [reply pentasharp/Controllers/BusSchedule.cs]: I would like to clarify that I had previously created the ApiStatusEnum enum in the Enums folder and the StandardResponse class in the DTOs folder, based on the recommendations you gave me earlier for the AuthenticateController.
However, I must admit that I forgot to follow that structure and logic in this specific feature I am currently working on, and I didn’t make use of what I had already created.

**Review by GentaSL — APPROVED** [review]

**Review by oltahasimja — APPROVED** [review]

**Pull request merged**